components/PoisList/PoisTable.vue

```diff
@@ -35,24 +34,26 @@ const search = ref('')
 // Fetch POIs by Cache or API
 const pois = ref()
 const loadingState = ref(false)
-const cachedKey = computed(() => `pois-${props.category.id}`)
-const { data: cachedPois } = useNuxtData(cachedKey.value)
 
-if (cachedPois.value) {
-  pois.value = cachedPois.value
-}
-else {
-  const { data, pending, error } = await useAsyncData(cachedKey.value, () => getPoiByCategoryId(
-    siteStore.config!,
-    props.category.id,
-    { geometry_as: 'point', short_description: true },
-  ))
-
-  if (error.value || !data.value)
-    throw createError({ statusCode: 404, statusMessage: 'POIs not found.' })
-
-  pois.value = data.value
-  loadingState.value = pending.value
+if (props.category) {
```

Comment:
Check for `!== undefined` and no  `props.category!.id` (`!`) latter?

Reply:
In here I think we are facing Typings limitations, even if I add `!== undefined` TS will shout that `props.category` is potentially `undefined` and we are within a `computed` or arrow function (which seems to break type inferance ...)

Reply:
providing empty color and icon is unexpected for TeritorioIconBadge, it shouldn't event accept that.

Reply:
Maybe use a default icon and color? eg `services-information`